test/nomad_main_test.go

```diff
@@ -0,0 +1,119 @@
+package test
+
+import (
+	"fmt"
+	"math/rand"
+	// "os"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/gcp"
+	"github.com/gruntwork-io/terratest/modules/packer"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+)
+
+const (
+	IMAGE_EXAMPLE_PATH = "../examples/nomad-consul-image/nomad-consul.json"
+	WORK_DIR           = "./"
+
+	// Terratest saved value names
+	SAVED_GCP_PROJECT_ID  = "GcpProjectId"
+	SAVED_GCP_REGION_NAME = "GcpRegionName"
+	SAVED_GCP_ZONE_NAME   = "GcpZoneName"
+
+	// Terraform module vars
+	TFVAR_NAME_GCP_PROJECT_ID = "gcp_project"
+	TFVAR_NAME_GCP_REGION     = "gcp_region"
+	TFVAR_NAME_GCP_ZONE       = "gcp_zone"
+
+	TFVAR_NAME_NOMAD_CONSUL_SERVER_CLUSTER_NAME = "nomad_consul_server_cluster_name"
+	TFVAR_NAME_NOMAD_CONSUL_SERVER_SOURCE_IMAGE = "nomad_consul_server_source_image"
+
+	TFVAR_NAME_NOMAD_CLIENT_CLUSTER_NAME = "nomad_client_cluster_name"
+	TFVAR_NAME_NOMAD_CLIENT_SOURCE_IMAGE = "nomad_client_source_image"
+
+	TFVAR_NAME_NOMAD_SERVER_CLUSTER_NAME = "nomad_server_cluster_name"
+	TFVAR_NAME_NOMAD_SERVER_SOURCE_IMAGE = "nomad_server_source_image"
+
+	TFVAR_NAME_CONSUL_SERVER_CLUSTER_NAME = "consul_server_cluster_name"
+	TFVAR_NAME_CONSUL_SERVER_SOURCE_IMAGE = "consul_server_source_image"
+
+	TFOUT_COLOCATED_SERVER_INSTANCE_GROUP_NAME = "nomad_server_instance_group_name"
+	TFOUT_SEPARATE_SERVER_INSTANCE_GROUP_NAME  = "nomad_server_instance_group_name"
+)
+
+type testCase struct {
+	Name string           // Name of the test
+	Func func(*testing.T) // Function that runs the test
+}
+
+var testCases = []testCase{
+	{
+		"TestDeployNomadCluster",
+		runNomadCluster,
```

Comment:
I guess we need to link up the right test methods here, btw do we need a testCases loop or could we get by with just separate test cases?

Reply:
The loop is because I'm building the ami just once, instead of once per testcase. Granted, we only have two test cases so I could just call these functions, really, but the number of test cases could grow in the future and we also use this pattern in other modules, so I thought why not

Reply:
Ok makes sense!